msu/hooks/skills/skill_container.nut

```diff
@@ -98,10 +98,44 @@
 
 	o.onMovementStep <- function( _tile, _levelDifference )
 	{
-		this.callSkillsFunction("onMovementStep", [
-			_tile,
-			_levelDifference
-		], false);
+		local wasUpdating = this.m.IsUpdating;
+		this.m.IsUpdating = true;
+		this.m.IsBusy = false;
+		this.m.BusyStack = 0;
+
+		local canMove = true;
+		local isRefunding = false;
+
+		foreach (skill in this.m.Skills)
+		{
+			if (!skill.isGarbage())
+			{
+				switch (_skill.onMovementStep(_tile, _levelDifference))
+				{
+					case true:
+						continue;
+
+					case false:
+						canMove = false;
+						break;
+
+					case null:
+						canMove = false;
+						isRefunding = true;
+						break;
+
+					default:
+						::logError("onMovementStep must return true, false, or null");
+						throw "invalid return value";
```

Comment:
Well this aint it. Rather have an Enum with the 3 states that can be returned. Or pass in a table where values can be changed by the function.

Reply:
Yeah this will be changed. @Enduriel wants something with partial refunds too, so it'll probably have to return a table or something.